### PR TITLE
Add TLS v1.2 support to EvseV2G module

### PR DESCRIPTION
**Pull request by FaHaGit**
Now the module supports TLS v1.2. The TLS support can be configured in the manifest.yaml file of the module. Session key logging can also be activated in the manifest.yaml file.

### CODE REVIEW

File: modules/EvseV2G/charger/ISO15118_chargerImpl.cpp [hunk 1]
```diff
@@ -47,6 +47,14 @@ void ISO15118_chargerImpl::init() {
...
         dlog(DLOG_LEVEL_DEBUG, "tls_security prohibit");
     }
+
+    v2g_ctx->tls_key_logging = mod->config.tls_key_logging;
+    v2g_ctx->tls_key_logging_path = mod->config.tls_key_logging_path;
+
+    if (mod->config.tls_key_logging == true) {
+        dlog(DLOG_LEVEL_DEBUG, "tls-key-logging enabled (path: %s)", mod->config.tls_key_logging_path.c_str());
+    }
+    v2g_ctx->certs_path = mod->config.certs_path;
```
Comment (SebaLukas) [inline modules/EvseV2G/charger/ISO15118_chargerImpl.cpp:57]: The certificates from the everest-core/config/certs folder are copied over to the everest-core/build/dist/etc/everest/certs with CMake during installation.

The path info.everest_prefix points to this everest-core/build/dist folder.

```cpp
const std::string EVEREST_ETC_CERTS_PATH = "etc/everest/certs";
v2g_ctx->certs_path = mod->info.everest_prefix + "/" + EVEREST_ETC_CERTS_PATH;
```

Please do not declare the const string in the function, but at the beginning of the file

File: README.md [hunk 2]
```diff
@@ -97,6 +97,8 @@ cd ~/checkout/everest-workspace/Josev/iso15118/shared/pki
 
 Note that this will generate an example PKI setup that can only be used for testing and simulation purposes. It will not work and is not recommended for production!
 
+The script for setting up PKI can also be used with the EvseV2G module, but make sure to place the generated PKI in the right path configured in the manifest.yaml.
```
Comment (SebaLukas) [inline README.md:100]: Possibly delete the second sentence, because it can lead to confusion.

File: modules/EvseV2G/tools.cpp [hunk 3]
```diff
@@ -329,3 +331,13 @@ uint8_t get_dir_numbered_file_names(char file_names[MAX_PKI_CA_LENGTH][MAX_FILE_
 
     return num_of_files + offset;
 }
+
+std::string convert_to_hex_str(const uint8_t* data, int len) {
+    std::stringstream sting_stream;
```
Comment (SebaLukas) [inline modules/EvseV2G/tools.cpp:336]: string_stream instead sting_stream

**Review by SebaLukas — CHANGES_REQUESTED** [review]
Looking good, only some small changes.
And a update to the current everest-core main.

**Comment by SebaLukas** [comment]
These codacy errors can be ignored

File: modules/EvseV2G/charger/ISO15118_chargerImpl.cpp (hunk 1 above)
Comment (FaHaGit) [reply modules/EvseV2G/charger/ISO15118_chargerImpl.cpp:57]: Where is this prefix defined in cmake file? I get "/usr" if I print out the string of mod->info.everest_prefix on charge byte platform.

File: modules/EvseV2G/manifest.yaml [hunk 4]
```diff
@@ -35,10 +35,29 @@ config:
...
+  tls_key_logging_path:
+    description:  >-
+      Output directory for the TLS key log file
+    type: string
+    default: /tmp
+  certs_path:
+    description: >-
+      Path to the V2G PKI
+    type: string
+    default: /etc/everest/certs
```
[the remark this answers is not in the document]
Comment (FaHaGit) [reply modules/EvseV2G/manifest.yaml:55]: This is currently used, I am wondering why it should not be used.

File: modules/EvseV2G/charger/ISO15118_chargerImpl.cpp (hunk 1 above)
Comment (FaHaGit) [reply modules/EvseV2G/charger/ISO15118_chargerImpl.cpp:57]: That would result to the file path "/usr/etc/everest/certs" which currently not exists.

Comment (SebaLukas) [reply modules/EvseV2G/charger/ISO15118_chargerImpl.cpp:57]: Here is my suggestion for a quick solution:
```cpp
const std::string EVEREST_ETC_CERTS_PATH = "etc/everest/certs";
[...]
if (mod->info.everest_prefix != "/usr") {
    v2g_ctx->certs_path = mod->info.everest_prefix + "/" + EVEREST_ETC_CERTS_PATH;
} else {
    v2g_ctx->certs_path = "/" + EVEREST_ETC_CERTS_PATH;
}
´´´

File: modules/EvseV2G/manifest.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (SebaLukas) [reply modules/EvseV2G/manifest.yaml:55]: No longer used when using the everest_prefix.

**Review by SebaLukas — APPROVED** [review]
👍

**Pull request merged**